en/middleware/xrce_dds.md

```diff
@@ -0,0 +1,335 @@
+   source install/local_setup.bash
+   ```
+1. Start the agent with settings for connecting to the XRCE-DDS client running on the simulator: 
+
+   ```sh
+   MicroXRCEAgent udp4 -p 8888
+   ```
+
+## Starting XRCE-DDS
+
+### Starting the Agent
+
+The agent is used to connect to the client over a particular channel, such as UDP, TCP, or a serial connection.
+The channel settings are specified when the agent is started, using command line options. 
+These are documented in the eProsima user guide: [Micro XRCE-DDS Agent > Agent CLI](https://micro-xrce-dds.docs.eprosima.com/en/latest/agent.html#agent-cli). <!-- what options work? i.e. does PX4 allow CAN? -->
+
+:::note
+You should create a single instance of the agent for each channel over which you need to connect.
+:::
+
+For example, the PX4 simulator runs the XRCE-DDS client over UDP on port 8888, so to connect to the simulator you would start the agent with the command:
+
+```sh
+MicroXRCEAgent udp4 -p 8888
+```
+
+On RaspberryPi, if you're using a serial link and connected to the IO pins, you might connect using this command:
+
+```sh
+sudo MicroXRCEAgent serial --dev /dev/AMA0 -b 921600
+```
+
+### Starting the Client
+
+The XRCE-DDS client module ([microdds-client](../modules/modules_system.md#microdds-client)) is included by default in all firmware and the simulator.
+This must be started with appropriate settings for the communication channel that you wish to use to communicate with the agent.
+
+On the simulator the client is automatically started on localhost UDP port 8888 using the default microdds namespace:
+
+```
+microdds_client start -t udp -p 8888
+```
+
+On flight controller hardware the client should be configured using the [Micro XRCE-DDS parameters](../advanced_config/parameter_reference.md#micro-xrce-dds):
+
+- [XRCE_DDS_0_CFG](../advanced_config/parameter_reference.md#XRCE_DDS_0_CFG): Set the port to connect on, such as `TELEM2`, `Ethernet`, or `Wifi`.
+  - [XRCE_DDS_UDP_PRT](../advanced_config/parameter_reference.md#XRCE_DDS_UDP_PRT):
+    If using an Ethernet connect, use this to specify the UDP port.
+  - [SER_TEL2_BAUD](../advanced_config/parameter_reference.md#SER_TEL2_BAUD), [SER_URT6_BAUD](../advanced_config/parameter_reference.md#SER_URT6_BAUD) (and so on):
+    If using a serial connection, the baud rate must be set using the `_BAUD` parameter associated with the serial port.
+    For example, you'd set a value for `SER_TEL2_BAUD` if you are connecting to the companion using `TELEM2`.
+    For more information see [Serial port configuration](../peripherals/serial_configuration.md#serial-port-configuration).
+
+  Some setups might also need these parameters to be set:
+
+  - [XRCE_DDS_KEY](../advanced_config/parameter_reference.md#XRCE_DDS_KEY): The XRCE-DDS key.
+    If you're working in a multi-client, single agent configuration, each client should have a unique key.
+  - [XRCE_DDS_DOM_ID](../advanced_config/parameter_reference.md#XRCE_DDS_DOM_ID): TBD.
```

Comment:
TBD could be added as a post process - but what is it for/how/when should it be set?

Reply:
- `XRCE_DDS_KEY` sets the `key` value of the client _session_. All clients connected to the same agent must have different keys (see https://micro-xrce-dds.docs.eprosima.com/en/stable/client_api.html#session , `uxr_init_session`). Moreover the key cannot have value zero. It is important to set this parameter only for multi vehicle setups and, more importantly, multi vehicle simulations, where all clients are connected in UDP to the same agent (multipla agents operating on different ports could be used but I don't se any advantages in doing that)

- `XRCE_DDS_DOM_ID` is the equivalent of the [`ROS_DOMAIN_ID`](https://docs.ros.org/en/foxy/Concepts/About-Domain-ID.html) and gives logical separation between DDS networks. Therefore clients configured with different `XRCE_DDS_DOM_ID` will belong to different networks. By default, ROS2 operates on ID 0